operator/pkg/ingress/cell.go

```diff
@@ -78,6 +83,11 @@ func registerReconciler(params ingressParams) error {
 		return nil
 	}
 
+	if params.Config.KubeProxyReplacement != option.KubeProxyReplacementTrue && !params.Config.EnableNodePort {
```

Comment:
@sayboras A couple of drive-by comments, as today I've been hit by this change on a work-in-progress CI workflow, to hear your thoughts before proposing any changes:

* Should we additionally account for the `KubeProxyReplacementStrict` case? I know that it has been marked as deprecated for a while now, but the Cilium CLI seems still pretty stubborn at [configuring it](https://github.com/cilium/cilium-cli/blob/590e75588a98b3855915b0d3f4b2957c04243b6b/install/autodetect.go#L221). Even configuring `--set kubeProxyReplacement=true` doesn't work, as the autoconfigured value still takes precedence. Using `--helm-set-string` works, but that's mostly by chance due to helm precedence rules. I can take a stab at fixing it tomorrow, but even if we release a fixed version users may still leverage older versions of the CLI for quite some time.
* Should the operator bail out hard if `EnableIngressController` is set, but the KPR requirements are not satisfied? The warning message is easy to miss, and can lead to surprises as users would expect ingresses to work, while they don't.

Reply:
Thinking a bit more about the first point, it is probably fine to ignore the deprecated setting as long as we fix the CLI soonish, given that we are still pretty early in the v1.16 development cycle, and users should have quite some time to upgrade the CLI before it gets released.